azuredevops/integration_suite_test.go

```diff
@@ -0,0 +1,183 @@
+	. "github.com/onsi/gomega"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+
+	"github.com/fluxcd/go-git-providers/gitprovider"
+)
+
+const (
+	azureTokenFile     = "/tmp/azure.token"
+	defaultDescription = "Foo description"
+)
+
+var (
+	azureDomain   = "dev.azure.com"
+	defaultBranch = "main"
+	testOrgName   = "flux"
+	testTeamName  = "fluxcd-test-team"
+	// placeholders, will be randomized and created.
+	testOrgRepoName string
+	testRepoName    string
+	client          gitprovider.Client
+)
+
+func init() {
+	// Call testing.Init() prior to tests.NewParams(), as otherwise -test.* will not be recognised. See also: https://golang.org/doc/go1.13#testing
+	testing.Init()
+	rand.Seed(time.Now().UnixNano())
+}
+
+func setupLogr() logr.Logger {
+	zapLog, err := setupLog()
+	if err != nil {
+		panic(fmt.Sprintf("who watches the watchmen (%v)?", err))
+	}
+	return zapr.NewLogger(zapLog)
+}
+
+func setupLog() (*zap.Logger, error) {
+	rawJSON := []byte(`{
+	  "level": "info",
+	  "encoding": "json",
+	  "outputPaths": ["stdout"],
+	  "errorOutputPaths": ["stderr"],
+	  "initialFields": {"name": "azure integration test"},
+	  "encoderConfig": {
+	    "messageKey": "message",
+	    "levelKey": "level",
+	    "levelEncoder": "lowercase"
+	  }
+	}`)
+
+	var cfg zap.Config
+	if err := json.Unmarshal(rawJSON, &cfg); err != nil {
+		return nil, err
+	}
+	logger, err := cfg.Build()
+	if err != nil {
+		return nil, err
+	}
+	defer logger.Sync()
+	return logger, nil
+}
+
+func TestProvider(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "azure Provider Suite")
+}
+
+var _ = Describe("azure Provider", func() {
+	var (
+		ctx context.Context = context.Background()
+	)
+
+	BeforeSuite(func() {
+
+		log := setupLogr()
+		log.V(1).Info("logger construction succeeded")
+
+		azureToken := os.Getenv("AZURE_TOKEN")
+		if azureToken == "" {
+			b, err := os.ReadFile(azureTokenFile)
+			if token := string(b); err == nil && token != "" {
+				azureToken = token
+			} else {
+				Fail("couldn't acquire AZURE_TOKEN env variable")
+			}
+		}
+
+		if azureDomainVar := os.Getenv("AZURE_DOMAIN"); azureDomainVar != "" {
+			azureDomain = azureDomainVar
+		}
+
+		if orgName := os.Getenv("GIT_PROVIDER_ORGANIZATION"); orgName != "" {
+			testOrgName = orgName
+		}
+
+		if teamName := os.Getenv("AZURE_TEST_TEAM_NAME"); teamName != "" {
+			testTeamName = teamName
+		}
+
+		var err error
+		client, err = NewClient(
+			azureToken,
+			ctx,
+			gitprovider.WithDomain(azureDomain),
+			gitprovider.WithDestructiveAPICalls(true),
+			gitprovider.WithConditionalRequests(true),
+			gitprovider.WithLogger(&log),
+		)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterSuite(func() {
+		if os.Getenv("SKIP_CLEANUP") == "1" {
+			return
+		}
+		// Don't do anything more if client wasn't created
+		if client == nil {
+			return
+		}
+
+		defer cleanupOrgRepos(ctx, "test-org-repo")
+		defer cleanupOrgRepos(ctx, "test-shared-org-repo")
```

Comment:
```suggestion
		cleanupOrgRepos(ctx, "test-org-repo")
		cleanupOrgRepos(ctx, "test-shared-org-repo")
```
`defer` is unnecessary here.